ccmlib/common.py

```diff
@@ -943,20 +943,25 @@ def assert_jdk_valid_for_cassandra_version(cassandra_version):
         exit(1)
 
 
-def aws_bucket_ls(s3_url):
+def aws_bucket_ls(s3_url, public=True):
     bucket_object = s3_url.replace('https://s3.amazonaws.com/', '').split('/')
     prefix = '/'.join(bucket_object[1:])
-    s3_conn = Session().client(service_name='s3', config=Config(signature_version=UNSIGNED))
+    config = Config(signature_version=UNSIGNED) if public else None
```

Comment:
the related document of boto3 does not explain what a `None` config implies. see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client . so i headed over to the source code of boto3, which led me to botocore. see https://github.com/boto/botocore/blob/db0aed3654ec7a5a1c64936d17a724ba41b2e926/botocore/session.py#L902 and https://github.com/boto/botocore/blob/db0aed3654ec7a5a1c64936d17a724ba41b2e926/botocore/session.py#L917-L922 . where i found that `default_client_config` is used if `config` is not specified or `None`. but `set_default_client_config()` is never called in this case. as `Session()` creates an instance of `Session` whose `_client_config` is None. see https://github.com/boto/botocore/blob/db0aed3654ec7a5a1c64936d17a724ba41b2e926/botocore/session.py#L151 .

so i am confused. what do you mean by 

> Support disabling `public` access for some buckets that
>  we don't have public access yet

if we don't have the public access to a certain bucket. shall we just give up or get authorized for accessing it?

i went ahead reading https://github.com/boto/botocore/blob/develop/botocore/session.py#L939-L957 . which led me to the credential resolver at https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L65. so i guess we rely on the environmental variables of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` set by scylla-pkg, am i right?

if that's true. could you please add a comment to explain the dependency of these assumption? because, to developers like me who is not familiar with boto3, it would be difficult to reason the connection between a `None` config and the environmental variables.

Reply:
`Config(signature_version=UNSIGNED)` means you'll use the API without using credentials at all
https://dev.to/aws-builders/access-s3-public-data-without-credentials-4f06

passing None, should fallback to defaults, as if you didn't passed that parameter, and after test, it's behaved as I would expect, I didn't read all the boto3/botocore to verify it... I've tested it.

Reply:
I've removed it completely it's not really was needed

Reply:
no, i am not questioning your change. i am just trying to find the official document to prove it.